cli/core/validator.go

```diff
@@ -2,17 +2,91 @@ package core
 	"math/big"
 	"strconv"
 
 	eigenpodproofs "github.com/Layr-Labs/eigenpod-proofs-generation"
 	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/core/onchain"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/fatih/color"
 )
 
-func GenerateValidatorProof(ctx context.Context, eigenpodAddress string, eth *ethclient.Client, chainId *big.Int, beaconClient BeaconClient) (*eigenpodproofs.VerifyValidatorFieldsCallParams, []uint64) {
+func SubmitValidatorProof(ctx context.Context, owner, eigenpodAddress string, chainId *big.Int, eth *ethclient.Client, batchSize uint64, proofs *eigenpodproofs.VerifyValidatorFieldsCallParams, noPrompt bool) ([]*types.Transaction, error) {
+
+	ownerAccount, err := PrepareAccount(&owner, chainId)
+	if err != nil {
+		return nil, err
+	}
+	PanicOnError("failed to parse private key", err)
+
+	eigenPod, err := onchain.NewEigenPod(common.HexToAddress(eigenpodAddress), eth)
+	if err != nil {
+		return nil, err
+	}
+
+	indices := Uint64ArrayToBigIntArray(proofs.ValidatorIndices)
+	validatorIndicesChunks := chunk(indices, batchSize)
+	validatorProofsChunks := chunk(proofs.ValidatorFieldsProofs, batchSize)
+	validatorFieldsChunks := chunk(proofs.ValidatorFields, batchSize)
+	if !noPrompt {
+		PanicIfNoConsent(fmt.Sprintf("This will call EigenPod.VerifyWithdrawalCredentials() %d times, to link your validator to your eigenpod.", len(validatorIndicesChunks)))
+	}
+
+	color.Green("calling EigenPod.VerifyWithdrawalCredentials() (using %d txn(s), max(%d) proofs per txn)", len(indices), batchSize)
+
+	latestBlock, err := eth.BlockByNumber(ctx, nil)
+	if err != nil {
+		return []*types.Transaction{}, err
+	}
+
+	transactions := []*types.Transaction{}
+	numChunks := len(validatorIndicesChunks)
+
+	color.Green("Submitting proofs with %d transactions", numChunks)
+
+	for i := 0; i < numChunks; i++ {
+		curValidatorIndices := validatorIndicesChunks[i]
+		curValidatorProofs := validatorProofsChunks[i]
+
+		var validatorFieldsProofs [][]byte = [][]byte{}
+		for i := 0; i < len(curValidatorProofs); i++ {
+			pr := curValidatorProofs[i].ToByteSlice()
+			validatorFieldsProofs = append(validatorFieldsProofs, pr)
+		}
+		var curValidatorFields [][][32]byte = CastValidatorFields(validatorFieldsChunks[i])
+
+		txn, err := SubmitValidatorProofChunk(ctx, ownerAccount, eigenPod, chainId, eth, curValidatorIndices, curValidatorFields, proofs, validatorFieldsProofs, latestBlock.Time())
+		if err != nil {
+			return transactions, err
+		}
+
+		transactions = append(transactions, txn)
+	}
+
+	return transactions, err
+}
+
+func SubmitValidatorProofChunk(ctx context.Context, ownerAccount *Owner, eigenPod *onchain.EigenPod, chainId *big.Int, eth *ethclient.Client, indices []*big.Int, validatorFields [][][32]byte, proofs *eigenpodproofs.VerifyValidatorFieldsCallParams, validatorFieldsProofs [][]byte, oracleBeaconTimesetamp uint64) (*types.Transaction, error) {
```

Comment:
just pass in the state root proof, not the entire`VerifyValidatorFieldsCallParams` imo, cleaner to also call it `stateRootProof` to differentiate from the million other proofs :)

Reply:
`SubmitCredentialProofChunk`?

yall are both so verbose :sweat:

Reply:
my brain hurts from these func signatures